### PR TITLE
Remove support for multiple tag types.

### PR DESCRIPTION
**Pull request by sebright**
This commit removes the class hierarchies for TagKey, TagValue, and Tag.  The
three classes are now concrete AutoValue classes that use Strings for keys and
values.  The serialization format isn't affected, because it already only
supported string tags.

________________________________________

/cc @bogdandrutu @adriancole

### CODE REVIEW

File: core/src/main/java/io/opencensus/tags/TagContextBuilder.java [hunk 1]
```diff
@@ -17,12 +17,6 @@
...
 import io.opencensus.common.Scope;
-import io.opencensus.tags.TagKey.TagKeyBoolean;
-import io.opencensus.tags.TagKey.TagKeyLong;
-import io.opencensus.tags.TagKey.TagKeyString;
-import io.opencensus.tags.TagValue.TagValueBoolean;
-import io.opencensus.tags.TagValue.TagValueLong;
-import io.opencensus.tags.TagValue.TagValueString;
 
 /** Builder for the {@link TagContext} class. */
 // TODO(sebright): Decide what to do when 'put' is called with a key that has the same name as an
```
Comment (songy23) [inline core/src/main/java/io/opencensus/tags/TagContextBuilder.java]: This TODO can be removed now.

File: core/src/main/java/io/opencensus/tags/TagContextBuilder.java [hunk 2]
```diff
@@ -36,25 +30,7 @@
    * @param value the value to set for the given key.
```
Comment (songy23) [inline core/src/main/java/io/opencensus/tags/TagContextBuilder.java]: nit: s/value/{@code TagValue}

Comment (sebright) [reply core/src/main/java/io/opencensus/tags/TagContextBuilder.java]: Done

File: core/src/main/java/io/opencensus/tags/TagContextBuilder.java (hunk 1 above)
Comment (sebright) [reply core/src/main/java/io/opencensus/tags/TagContextBuilder.java]: :+1:

**Comment by bogdandrutu** [comment]
@dinooliva can you approve this if you think it is fine?

**Comment by sebright** [comment]
I'll hold off merging this until we're sure that we don't need the other types of tags.

**Comment by adriancole** [comment]
Fyi from a serialization pov. I know jonah from appD mentioned they have a
separate header for numeric tags

**Comment by sebright** [comment]
Thanks!

**Pull request merged**